From 1f90bcefd80187ed7b3bf9f953a631e715c539a1 Mon Sep 17 00:00:00 2001
From: Yutao Yuan <infmagic2047reg@outlook.com>
Date: Sun, 20 Aug 2023 02:31:04 +0800
Subject: [PATCH] remove __rtruediv__ definitions which fail to compile with
 Cython 3

They fail to compile with Cython 3 and c_api_binop_methods=True, and are
never treated as special arithmetic methods by old Cython anyway, so
they can be safely removed.
---
 src/sage/rings/complex_mpc.pyx  | 20 --------------------
 src/sage/rings/complex_mpfr.pyx | 22 ----------------------
 2 files changed, 42 deletions(-)

Index: sage-test/sage/src/sage/rings/complex_mpc.pyx
===================================================================
--- sage-test.orig/sage/src/sage/rings/complex_mpc.pyx
+++ sage-test/sage/src/sage/rings/complex_mpc.pyx
@@ -1572,26 +1572,6 @@ cdef class MPComplexNumber(sage.structur
         mpc_norm(x.value, self.value, (<RealField_class>x._parent).rnd)
         return x
 
-    def __rtruediv__(self, left):
-        r"""
-        Returns the quotient of ``left`` with ``self``, that is: ``left/self``
-        as a complex number.
-
-        INPUT:
-
-        - ``left`` -- a complex number
-
-        EXAMPLES::
-
-            sage: MPC = MPComplexField()
-            sage: a = MPC(2, 2)
-            sage: a.__rtruediv__(MPC(1))
-            0.250000000000000 - 0.250000000000000*I
-            sage: MPC(1)/a
-            0.250000000000000 - 0.250000000000000*I
-        """
-        return MPComplexNumber(self._parent, left)/self
-
     def __pow__(self, right, modulus):
         """
         Compute ``self`` raised to the power of exponent, rounded in
Index: sage-test/sage/src/sage/rings/complex_mpfr.pyx
===================================================================
--- sage-test.orig/sage/src/sage/rings/complex_mpfr.pyx
+++ sage-test/sage/src/sage/rings/complex_mpfr.pyx
@@ -1671,28 +1671,6 @@ cdef class ComplexNumber(sage.structure.
         mpfr_clear(right_nm)
         return x
 
-    def __rtruediv__(self, left):
-        r"""
-        Return the quotient of left with ``self``, that is:
-
-        ``left/self``
-
-        as a complex number.
-
-        INPUT:
-
-        - ``left`` - a complex number to divide by ``self``
-
-        EXAMPLES::
-
-            sage: a = ComplexNumber(2,0)
-            sage: a.__rtruediv__(CC(1))
-            0.500000000000000
-            sage: CC(1)/a
-            0.500000000000000
-        """
-        return ComplexNumber(self._parent, left)/self
-
     def __pow__(self, right, modulus):
         r"""
         Raise ``self`` to the ``right`` exponent.
